src/mpid/ch4/shm/src/shm_control.c

```diff
@@ -1,41 +0,0 @@
-/* -*- Mode: C; c-basic-offset:4 ; -*- */
-/*
- * (C) 2019 by Argonne National Laboratory.
- *     See COPYRIGHT in top-level directory.
- */
-
-#include "mpidimpl.h"
-#include "shm_control.h"
-#ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
-#include "../xpmem/xpmem_control.h"
-#endif
-
-int MPIDI_SHM_ctrl_dispatch(int ctrl_id, void *ctrl_hdr)
-{
-    int mpi_errno = MPI_SUCCESS;
-
-    switch (ctrl_id) {
-#ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
```

Comment:
Removing these functions will break #4331. We need to figure out what to do with that PR before merging this one I think.

Reply:
No problem. Both PRs probably need be looked together at some point.

Reply:
We decided to close #4331 so this comment is irrelevant now.

Reply:
Cool. @wesbland if you get a chance, could you review this PR anyway?

Reply:
I actually already looked at it and this it's generally fine, but I wanted to see the resolution of @minsii 's comments since she had some opinions that were probably more helpful than mine.